### ISSUE TITLE
Date range facet widgets fail form label and contrast accessibility checks

### ISSUE THREAD

**Issue by patdunlavey** [esmero/format_strawberryfield#538]
We are seeing form label errors and low contrast errors on the "Format Strawberryfield Date Range slider" (sbf_range_date_slider):
<img width="869" height="586" alt="Image" src="https://github.com/user-attachments/assets/eef12a3f-33e6-495e-a7d0-695fe9ca99b2" />

Presumably we are seeing similar errors on the "Format Strawberryfield Date Range Picker (No Slider)" (sbf_date_range) as well.

The form label errors - an absence of a "for" attribute on the label elements - is caused by the form render arrays not including an ['#id'] value for the labeled form elements. (['#attributes']['id'] does not suffice, apparently).

The low contrast errors come from jquery-ui-slider-pips. That library is no longer being maintained, so the simplest solution, I think, is to provide overriding css for those slider labels.

I have patched the format_strawberryfield module (1.5.x-dev) to fix the form label errors and can offer a PR for that. 

I fixed the low contrast errors in my theme's css, but can re-roll that into a PR for the format_strawberryfield module as well, if you like! (format_strawberryfield/modules/format_strawberryfield_facets/css/slider.css)

**Comment by DiegoPino** [comment]
@patdunlavey thanks for bringing these accessibility issues! Few questions

are the  "for" attribute on the label elements a responsibility of the underlaying theme (templates) handling each element or on us? If on us happy to provide solutions. The Fixed IDs (#id) might be an issue at the theme level or render array (solvable). There is a trend of moving out of fixed IDS (bc of Ajax) into data attributes.

Low contrast can be solved, true, via CSS overrides and so happy and grateful to get contributions for that from you (please against 1.6.0)! Pull away!

Happy to code for a different slider JS, if you have a proposal, I could add an "config option" that works around that.

**Comment by patdunlavey** [comment]
@DiegoPino I don't know how other themes work, but the bootstrap_barrio (base) theme expects the ['#id'] value as I described. Its form-element.html.twig file utilizes `{{ label }}` in some places, so I believe that fixing it at the theme level would entail render array bubbling nonsense. I spent a while trying to do exactly that, and gave up because I couldn't work out how to fix it in either a preprocess function or in a template (without completely hacking the template). Maybe I missed some easy trick? 

Will using `Html::getUniqueId()` keep us ajax-safe?

I'm currently composer requiring 1.5.x-dev and I don't think I'm ready to start running 1.6.0 on a prod site yet. I'm happy to roll a PR against 1.6.0, but I'm not set up to test it. It looks like there's zero diff between 1.5.0 and 1.6.0 for the three affected files though: 
- modules/format_strawberryfield_facets/src/Plugin/facets/widget/DateRangeSliderWidget.php
- modules/format_strawberryfield_facets/src/Plugin/facets/widget/DateRangeWidget.php
- modules/format_strawberryfield_facets/css/slider.css

**Comment by patdunlavey** [comment]
PR against 1.6.0 created: https://github.com/esmero/format_strawberryfield/pull/539

**Comment by DiegoPino** [comment]
Resolved via https://github.com/esmero/format_strawberryfield/commit/e3df9f0a44446d9ab7c498b0099b0776e5e3134f. Thanks @patdunlavey !!
